client/post-editor/media-modal/index.jsx

```diff
@@ -111,7 +111,13 @@ export class EditorMediaModal extends Component {
 	isDisabled() {
 		return some( this.props.mediaLibrarySelectedItems, function( item ) {
 			const mimePrefix = MediaUtils.getMimePrefix( item );
-			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.props.view );
+			return item.transient && (
+				// Transients can't be handled by the editor if they are being
+				// uploaded via an external URL
+				item.is_uploading_via_url ||
```

Comment:
I dunno, I actually think it's more direct to test for the presence of the value that would actually be used in generating the image. The issue is that we're trying to show an image in the editor, but the `media.URL` is undefined so it appears broken. If we know whether we'd be able to insert anything by checking `media.URL`, we could determine ahead of time whether it wouldn't be a good item to use. The downside I see to this approach is that it may mislead a developer to think this is a real property of the media endpoints response payload.